agent/internal/actor/http.go

```diff
@@ -37,8 +38,14 @@ func NewIPActionHTTPHandler(action Action, ip net.IP) (http.Handler, error) {
 
 // NewUserActionHTTPHandler returns a HTTP handler that should be applied at
 // the request handler level to perform the security response.
-func NewUserActionHTTPHandler(action Action, userID map[string]string) http.Handler {
-	return newBlockHTTPHandler(blockUserEventName, newBlockedUserEventProperties(action, userID))
+func NewUserActionHTTPHandler(action Action, userID map[string]string) (http.Handler, error) {
+	switch actual := action.(type) {
+	case blockAction:
+		return newBlockHTTPHandler(blockUserEventName, newBlockedUserEventProperties(actual, userID)), nil
+	case *redirectAction:
```

Comment:
Weird that there's a pointer and a non-pointer here.

Reply:
That's a small Go optimization: such wrapper type of a pointer type can be used as a value because only the pointer value will be copied when passed - it also allows to save some GC time and not allocate yet another structure to be monitored.